salt/states/github.py

```diff
@@ -583,26 +598,134 @@ def repo_present(
+            # before continuing.
+            fetched = False
+            for attempt in range(3):
+                time.sleep(1)
+                try:
+                    __salt__['github.get_repo_teams'](
+                        name, profile=profile, **kwargs)
+                    fetched = True
+                    break
+                except CommandExecutionError as e:
+                    log.info("Attempt {0} to fetch new repo {1} failed".format(
+                        attempt, name))
+
+            if not fetched:
+                ret['result'] = False
+                ret['comment'] = 'Failed to verify repo {0} after creation.'.format(name)
+                return ret
 
-        add_params = dict(given_params)
-        add_params.update(kwargs)
-        result = __salt__['github.add_repo'](
-            name,
-            **add_params
-        )
-        if result:
             ret['changes']['repo'] = {
                 'old': None,
                 'new': 'Repo {0} has been created'.format(name)
             }
+
+    if teams is not None:
+        if __opts__['test'] and not target:
+            # Assume no teams if we're in test mode and the repo doesn't exist
+            current_teams = []
         else:
-            ret['result'] = False
-            ret['comment'] = 'Failed to create repo {0}.'.format(name)
-            return ret
+            current_teams = __salt__['github.get_repo_teams'](name, profile=profile)
+        current_team_names = set([t['name'] for t in current_teams])
 
+        # First remove any teams that aren't present
+        for team_name in current_team_names:
+            if team_name not in teams:
+                if __opts__['test']:
+                    test_comments.append('Repo {0} is set to be removed from team {1}.'
+                                         .format(name, team_name))
+                    ret['result'] = None
+                else:
+                    result = __salt__['github.remove_team_repo'](name, team_name,
+                                                                 profile=profile)
+                    if result:
+                        ret['changes'][team_name] = {
+                            'old': 'Repo {0} is in team {1}'.format(name, team_name),
+                            'new': 'Repo {0} is not in team {1}'.format(name, team_name)
+                        }
+                    else:
+                        ret['result'] = False
+                        ret['comment'] = ('Failed to remove repo {0} from team {1}.'
+                                          .format(name, team_name))
+                        return ret
+
+        # Next add or modify any necessary teams
+        for team_name, permission in six.iteritems(teams):
+            if team_name not in current_team_names:  # Need to add repo to team
+                if __opts__['test']:
+                    test_comments.append('Repo {0} is set to be added to team {1}.'
```

Comment:
I have the same comment for this dictionary as well that I commented about above with the new/old dict.